### ISSUE TITLE
Reconcile POM.compareVersions and VersionUtils.compare

### ISSUE THREAD

**Issue by stelfrich** [scijava/scijava-common#334]
The two implementations ([`POM.compareVersions()`](https://github.com/scijava/scijava-common/blob/9554f8f35f6aa1319f7b02587e266064a3873181/src/main/java/org/scijava/util/POM.java#L298-L359) and [`VersionUtils.compare()`](https://github.com/scijava/scijava-common/blob/9554f8f35f6aa1319f7b02587e266064a3873181/src/main/java/org/scijava/util/VersionUtils.java#L117-L135)) are semantically identical and should be reconciled.

- [ ] Deprecate `POM.compareVersions()`
- [ ] Use `VersionUtils.compare()` in `POM.compareTo()`
- [ ] Keep Javadoc of `POM.compareVersions()`
- [ ] Keep implementation of `VersionUtils.compare()`

If you agree, @ctrueden, you can assign me and I can work on it.

**Comment by ctrueden** [comment]
I agree in broad strokes. Let's wait a couple of weeks until have the scaffold for SciJava 3 modularization in place. Then we can do the reconciliation there.

**Comment by stelfrich** [comment]
Agreed! :+1:
